test/Quake/memtoreg-2.qke

```diff
@@ -0,0 +1,811 @@
+  return
+}
+
+func.func @apply_parametrized_one_target_operators() {
+  %angle = arith.constant 3.14 : f64
+  %q0 = quake.alloca  !quake.ref
+  %q1 = quake.alloca  !quake.ref
+  quake.r1 (%angle) [%q1] %q0 : (f64, !quake.ref, !quake.ref) -> ()
+  quake.rx (%angle) [%q1] %q0 : (f64, !quake.ref, !quake.ref) -> ()
+  quake.ry (%angle) [%q1] %q0 : (f64, !quake.ref, !quake.ref) -> ()
+  quake.rz (%angle) [%q1] %q0 : (f64, !quake.ref, !quake.ref) -> ()
+  return
+}
+
+func.func @qextract_and_apply_controlled_two_targets_operator() {
+  %c_0 = arith.constant 0 : i32
+  %c_1 = arith.constant 1 : i32
+  %c_2 = arith.constant 2 : i32
+  %0 = quake.alloca !quake.veq<3>
+  %q0 = quake.extract_ref %0 [%c_0] : (!quake.veq<3>,i32) -> !quake.ref
+  %q1 = quake.extract_ref %0 [%c_1] : (!quake.veq<3>,i32) -> !quake.ref
+  %q2 = quake.extract_ref %0 [%c_2] : (!quake.veq<3>,i32) -> !quake.ref
+  quake.swap [%q2] %q0, %q1 : (!quake.ref,!quake.ref,!quake.ref) -> ()
+  return
+}
+
+func.func @mz_and_reset_veq_with_extracted_refs() {
+  %c_0 = arith.constant 0 : i32
+  %c_1 = arith.constant 1 : i32
+  %0 = quake.alloca !quake.veq<2>
+  %q0 = quake.extract_ref %0[%c_0] : (!quake.veq<2>, i32) -> !quake.ref
+  %q1 = quake.extract_ref %0[%c_1] : (!quake.veq<2>, i32) -> !quake.ref
+  %reg = quake.mz %0 : (!quake.veq<2>) -> !cc.stdvec<i1>
+  quake.reset %0 : (!quake.veq<2>) -> ()
+  return
+}
+
+func.func @simple_loop() {
+  %c0_i64 = arith.constant 0 : i64
+  %c1_i64 = arith.constant 1 : i64
+  %c42_i64 = arith.constant 42 : i64
+  %q0 = quake.alloca !quake.ref
+  // memtoreg does not promote memref types.
+  %alloca = memref.alloca() : memref<i64>
+  memref.store %c0_i64, %alloca[] : memref<i64>
+  cc.loop while {
+    %1 = memref.load %alloca[] : memref<i64>
+    %2 = arith.cmpi ult, %1, %c42_i64 : i64
+    cc.condition %2
+  } do {
+    quake.x %q0 : (!quake.ref) -> ()
+    cc.continue
+  } step {
+    %1 = memref.load %alloca[] : memref<i64>
+    %2 = arith.addi %1, %c1_i64 : i64
+    memref.store %2, %alloca[] : memref<i64>
+  }
+  quake.z %q0 : (!quake.ref) -> ()
+  return
+}
+
+func.func @floop_with_vector_and_qextract() {
+  %c0_i64 = arith.constant 0 : i64
+  %c1_i64 = arith.constant 1 : i64
+  %c2_i64 = arith.constant 2 : i64
+  %veq = quake.alloca  !quake.veq<2>
+  %q0 = quake.extract_ref %veq[%c0_i64] : (!quake.veq<2> ,i64) -> !quake.ref
+  %alloca = memref.alloca() : memref<i64>
+  memref.store %c0_i64, %alloca[] : memref<i64>
+  cc.loop while {
+    %3 = memref.load %alloca[] : memref<i64>
+    %4 = arith.cmpi ult, %3, %c2_i64 : i64
+    cc.condition %4
+  } do {
+    %3 = memref.load %alloca[] : memref<i64>
+    %4 = quake.extract_ref %veq[%3] : (!quake.veq<2> ,i64) -> !quake.ref
+    cc.continue
+  } step {
+    %3 = memref.load %alloca[] : memref<i64>
+    %4 = arith.addi %3, %c1_i64 : i64
+    memref.store %4, %alloca[] : memref<i64>
+  }
+  %2 = quake.mz %veq : (!quake.veq<2>) -> !cc.stdvec<i1>
+  return
+}
+
+// CHECK-LABEL:   func.func @classical_if01(
```

Comment:
Can we intersperse these FileCheck sections within the above code, so that the tests are close to the code?